### PR TITLE
Added styling event, option and scope attribute

### PR DESCRIPTION
**Pull request by chaosmail**
Hi Sébastien,

here is what I propose for styling: an event on it's own, a scope attribute that will be passed to the event handlers. I just implemented the first step, to show you what I mean. Now we could pass a `styles` dict (which could maybe defined by a string in the options, like `style: default`) to the styles handler and use it in every factory. Please let me know, what you think about this.

Best,
Christoph

### CODE REVIEW

**Comment by chaosmail** [comment]
ok I made a rebase, and there are just some small changes left ;)

**Comment by chaosmail** [comment]
I added a `Factory.StyleSheet` that should apply the proper style based on the options. Maybe now, we can make all styles more lightweight.

File: src/factories/StyleSheet.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module n3Charts.Factory {
+  'use strict';
+
+  export class StyleSheet extends Utils.BaseFactory {
+
+    public style: Utils.IFactory;
+
+    update(datasets, options, attributes: ng.IAttributes) {
+      // Load a style
+      this.style = new Style.DefaultStyle();
```
Comment (lorem--ipsum) [inline src/factories/StyleSheet.ts]: Hm this looks like it belongs to the `create` method, don't you think so ?

**Comment by lorem--ipsum** [comment]
:poultry_leg:

**Pull request merged**